torchbenchmark/score/compute_score.py

```diff
@@ -9,10 +9,42 @@
 import os
 import yaml
 
+from collections import defaultdict
 from tabulate import tabulate
+from torchbenchmark import list_models
 
-SPEC_FILE_DEFAULT = "score.yml"
-TARGET_SCORE_DEFAULT = 1000
+def generate_spec():
```

Comment:
i'm kind of wondering if adding this helper makes things any cleaner.  I guess at least the spec file doesn't have to exist... but i think we can still do better.

Given that you are essentially creating an empty data structure here based on the models that exist and are expected, could you do that just as easily inside generate_bench_config?

Reply:
In short we can. But I am kind of confused. Even if we move , that would mean we either bake this code inside `generate_bench_config` or add a call to this helper inside `generate_bench_config`.
```
if spec is None:
   spec = generate_spec()
```

Reply:
i'm looking for a bigger cleanup than just moving the helper.  

At at a first order, I was thinking that maybe there is no point in creating the dictionary, only to then use it as a way to drive the nested for loops starting at L61 here:
https://github.com/pytorch/benchmark/blob/master/torchbenchmark/score/generate_score_config.py#L61

So, you could just try to clean up how generate_score_config works.

But.. thinking even bigger, do we need the 'generate_score_config' function at all?  or is there a cleaner way to do the whole thing?  

All the user really wants to do is run 'compute_score', but we need it to be accurate and have the safety checks somehow.